File: backend/app/models/recurring_service.rb
class RecurringService < ApplicationRecord
  belongs_to :project

  enum :billing_cycle, { monthly: 0, yearly: 1, unique: 2 }
  enum :status, { active: 0, paused: 1 }

  validates :name, presence: { message: "El nombre del servicio es requerido" }
  validates :amount, presence: { message: "El monto es requerido" },
                     numericality: { greater_than: 0, message: "El monto debe ser mayor a 0" }
  validates :billing_cycle, presence: true
  validates :status, presence: true

  def self.ransackable_attributes(_auth_object = nil)
    %w[id name amount billing_cycle next_billing_date status project_id created_at updated_at]
  end

  def self.ransackable_associations(_auth_object = nil)
    %w[project]
  end
end

File: frontend/src/types/recurringService.ts
export interface RecurringService {
  id: number;
  project_id: number;
  project_name: string;
  client_name: string;
  name: string;
  amount: number;
  billing_cycle: string;
  next_billing_date: string | null;
  status: string;
  created_at: string;
  updated_at: string;
}

export interface RecurringServiceFormData {
  project_id: number | string;
  name: string;
  amount: number | string;
  billing_cycle: string;
  next_billing_date: string;
  status: string;
}

export const BILLING_CYCLES = [
  { value: "monthly", label: "Mensual" },
  { value: "yearly", label: "Anual" },
  { value: "unique", label: "Único" },
] as const;

export const SERVICE_STATUSES = [
  { value: "active", label: "Activo" },
  { value: "paused", label: "Pausado" },
] as const;
